src/transformers/models/vibevoice/generation_vibevoice.py

```diff
@@ -0,0 +1,765 @@
+            except queue.Empty:
+                pass
+
+        self.active_samples -= samples_to_remove
+
+        if batch_chunks:
+            return batch_chunks
+        elif self.active_samples:
+            # If no chunks were ready but we still have active samples, wait a bit and try again
+            time.sleep(0.01)
+            return self.__next__()
+        else:
+            raise StopIteration()
+
+
+class VibeVoiceGenerationMixin(GenerationMixin):
+    def _get_stopping_criteria(
+        self,
+        *args,
+        **kwargs,
+    ) -> StoppingCriteriaList:
+        criteria = super()._get_stopping_criteria(*args, **kwargs)
+
+        kept_criteria = StoppingCriteriaList()
+        for criterion in criteria:
+            if not isinstance(criterion, MaxLengthCriteria):
+                if isinstance(criterion, EosTokenCriteria):
+                    logger.debug(
+                        f"VibeVoice handles EOS tokens internally, ignoring {criterion.__class__.__name__} stopping criteria."
+                    )
+                else:
+                    logger.warning(
+                        f"VibeVoice does not support {criterion.__class__.__name__} stopping criteria, it will be ignored."
+                    )
+            else:
+                kept_criteria.append(criterion)
+        return kept_criteria
+
+    def _prepare_generation_config(
+        self, generation_config: Optional[GenerationConfig], use_model_defaults: Optional[bool] = None, **kwargs
+    ):
+        """
+        This method overrides [~generation.utils.GenerationMixin._prepare_generation_config].
+
+        It extracts VibeVoice-specific parameters from kwargs and sets up the default noise scheduler (if not provided).
+
+        VibeVoice-specific parameters include:
+        - `noise_scheduler`: An optional noise scheduler instance to use instead of the default.
+        - `cfg_scale`: A classifier-free guidance scale to use during generation.
+        - `n_diffusion_steps`: Number of diffusion steps to use during generation of each audio chunk.
+        - `monitor_progress`: A callable to monitor generation progress. If provided, this function can be called to
+            report the progress of the audio generation. The function takes a tensor argument `p` of shape `(n, 3)`,
+            where `n` is the batch size. `p[i, 0]` contains the current generation step for batch item `i`, `p[i, 1]`
+            contains the maximum generation steps for batch item `i` (which may vary based on input length), and
+            `p[i, 2]` contains the actual completion step for finished samples. No return value is expected.
+
+        """
+        noise_scheduler = kwargs.pop("noise_scheduler", None)
+        if "max_new_tokens" in kwargs and kwargs["max_new_tokens"] is None:
+            # pop to default to generation_config max_length behavior instead of internal default of 20
+            kwargs.pop("max_new_tokens")
```

Comment:
Added so that `_prepare_generation_config` doesn't use the default of 20 max_new_tokens, even if it's set in the generation config ([here](https://huggingface.co/bezzam/VibeVoice-1.5Bv2/blob/main/generation_config.json#L4))